### PR TITLE
Allow environment_file in YAML files

### PR DESCRIPTION
**Pull request by alexellis**
Signed-off-by: Alex Ellis <alexellis2@gmail.com>

## Description

Allow environmental variables to be defined in an external file

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/alexellis/faas/blob/master/CONTRIBUTING.md)) (pending)

Lowers risk of inadvertently committing secrets to Git.

## How Has This Been Tested?

Tested with sample app and config:

env_files.yml
```
provider:
  name: faas
  gateway: http://localhost:8080

functions:
  env_files:
    lang: node
    handler: ./env_files
    image: env_files
    environment:
     url: http://hyper.sh
     secret_key: lolz
    environment_file:
     - ./login.yml
     - ./urls.yml
```

login1.yml
```
environment:
  account: alex.ellis
  secret_key: dogfight
```

urls.yml

```
environment:
  url: https://hyper.sh
  post_back: https://b.z.y
```

handler.js
```
"use strict"

module.exports = (context, callback) => {
    callback(undefined, process.env);
}
```

Sample function prints a map of environmental variables:

```
  account: 'alex.ellis',
  post_back: 'https://b.z.y',
  secret_key: 'dogfight', 
  url: 'https://hyper.sh',
```

See the changes to the README for spec/docs.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,7 +111,29 @@ $ faas-cli deploy -f ./samples.yml
 
 ### Managing secrets
 
-You can deploy secrets and configuration via environmental variables.
+You can deploy secrets and configuration via environmental variables in-line or via external files.
+
+> Note: external files take priority over in-line environmental variables. This allows you to specify a default and then have overrides within an external file.
```
Comment (johnmccabe) [inline README.md]: worth making the priority clear when the same envvar appears in multiple files, is it the first file loaded, the last etc?

**Comment by johnmccabe** [comment]
Just one comment, apart from that it looks good to me.

**Comment by alexellis** [comment]
Agreed we should say it's a deterministic order

**Comment by alexellis** [comment]
Updates made. Merging.

**Comment by johnmccabe** [comment]
👍

**Pull request merged**